### PR TITLE
Devices list

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-commons/lib/core/models/device.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface ConnectedDevice {
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/models/device.ts]: Let's just use `DeviceInfo`

File: universal-login-react/src/ui/Settings/ManageDevices.tsx [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
-  <Accordion
-    title="Manage devices"
-    subtitle="You currently have 3 authorized devices"
-  >
-    <div>
-      <button className="settings-btn">Remove this device</button>
-    </div>
-  </Accordion>
-);
+interface ConnectedDevice {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Settings/ManageDevices.tsx]: Unnecessary

File: universal-login-react/src/ui/Settings/ManageDevices.tsx [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+interface ManageDevicesProps {
+  sdk: UniversalLoginSDK;
+  privateKey: string;
+  contractAddress: string;
+}
+
+export const ManageDevices = ({sdk, contractAddress, privateKey}: ManageDevicesProps) => {
+  const [devices, setDevices] = useState<ConnectedDevice[]>([]);
+  useEffect(() => sdk.subscribeConnectedDevices(contractAddress, privateKey, setDevices), []);
+  console.log(devices);
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Settings/ManageDevices.tsx]: console.log...

File: universal-login-react/src/ui/Settings/ManageDevices.tsx (hunk 2 above)
Comment (xitronix) [reply universal-login-react/src/ui/Settings/ManageDevices.tsx]: ✔️

File: universal-login-commons/lib/core/models/device.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-commons/lib/core/models/device.ts]: ✔️

File: universal-login-react/src/ui/Settings/ManageDevices.tsx [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
+interface ManageDevicesProps {
+  devices?: DeviceInfo[];
+}
+
+export const ManageDevices = ({devices}: ManageDevicesProps) => (
   <div>
-    <button className="settings-btn">Remove this device</button>
-  </div>
+    {
+      devices && devices.map(({os}: DeviceInfo, index: number) => (
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Settings/ManageDevices.tsx]: I generally prefer default values for parameters and avoiding patterns like `x && x.map(...)`

File: universal-login-react/src/ui/Settings/ManageDevices.tsx [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
+        <div key={index}>
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Settings/ManageDevices.tsx]: `index` is probably a bad key, hmmm `publicKey` would be useful afterall :(

Comment (xitronix) [reply universal-login-react/src/ui/Settings/ManageDevices.tsx]: I will replace it in next PR

File: universal-login-react/src/ui/Settings/ManageDevices.tsx (hunk 4 above)
Comment (xitronix) [reply universal-login-react/src/ui/Settings/ManageDevices.tsx]: Replaced by conditional rendering info that devices are loading

**Pull request merged**